site/src/pages/[shortname]/blocks/[blockSlug].page.tsx

```diff
@@ -35,15 +35,36 @@ const blockRequire = (name: BlockDependency) => {
   return blockDependencies[name];
 };
 
-const blockEval = (source: string): BlockExports => {
+const blockComponentFromSource = (source: string): ComponentType => {
+  // this does not guarantee filtering out all non-React components
+  if (!source.includes("createElement")) {
+    throw new Error(
+      "Block is not a React component - please implement rendering support for whatever it is and update this check. Note that we may in future change the recommended implementation for blocks to avoid this issue.",
```

Comment:
@CiaranMn Within this error we should provide a link to whichever best public thread (GitHub Issue, Discussion or other) we have that touches on this, in case someone wants to join the conversation. This is a good point to capture them.